File: app/src/main/java/org/mesonet/app/formulas/MathMethods.java
package org.mesonet.app.formulas;


import javax.inject.Inject;

public class MathMethods
{
    @Inject
    MathMethods(){}



    public Number Round(Number inBase, int inDecimals)
    {
        if(inBase == null)
            return null;

        double tenTimesDecimals = Math.pow(10, inDecimals);

        return Math.round(inBase.doubleValue() * tenTimesDecimals) / tenTimesDecimals;
    }
}

File: dataprocessing/src/main/java/org/mesonet/dataprocessing/formulas/MathMethods.kt
package org.mesonet.dataprocessing.formulas


import javax.inject.Inject

class MathMethods @Inject
internal constructor() {


    fun Round(inBase: Number?, inDecimals: Int): Number? {
        if (inBase == null)
            return null

        val tenTimesDecimals = Math.pow(10.0, inDecimals.toDouble())

        return Math.round(inBase.toDouble() * tenTimesDecimals) / tenTimesDecimals
    }
}
